app/src/main/java/com/jwisozk/flightreviews/AbsParamCell.kt

```diff
@@ -0,0 +1,24 @@
+package com.jwisozk.flightreviews
+
+import androidx.annotation.NonNull
+
+abstract class AbsParamCell(@NonNull private val _viewType: ViewType) {
+
+    enum class ViewType {
+        RATING_BAR,
+        EDIT_TEXT,
+        SUBMIT_BUTTON
+    }
```

Comment:
I've noticed that inheritors of `AbsParamCell` hold a reference to `*ViewModel` / `*Listener`. That's a broken implementation. Cell classes shouldn't know anything about `ViewModel` or any kind of listener. They hold data necessary to draw something and probably some id (`LabelType` in this case is kind of an id) to handle clicks and other actions. 
I guess this `AbsParamCell.ViewType` pattern comes from our Java guides. But we have better ways to express this in Kotlin. Here is how this whole thing should look like:
```kotlin
sealed class ParamCell {

    data class RatingBar(
        val labelType: AbsParamCell.LabelType,
        val rating: Float,
        val isCheckboxVisible: Boolean,
        val isCheckboxChecked: Boolean
    ) : ParamCell()

    data class EditText(val text: String) : ParamCell()
    object Button : ParamCell()
}

abstract class AbsParamHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    abstract fun bind(cell: ParamCell)
}

class RatingBarParamHolder(
    inflater: LayoutInflater,
    parent: ViewGroup,
    listener: ParamAdapter.Listener
) : AbsParamHolder(inflater.inflate(R.layout.rating_bar, parent, false)) {

    private val ratingBarChangeListener = RatingBar.OnRatingBarChangeListener { _, rating, _ ->
        cell?.let {
            listener.onRatingBarChangeListener(it, rating)
        }
    }
    private val checkedChangeListener = CompoundButton.OnCheckedChangeListener { _, isChecked ->
        cell?.let {
            listener.onCheckedChangeListener(it, isChecked)
        }
    }

    private val ratingBar = itemView.findViewById<RatingBar>(R.id.ratingBar).apply {
        onRatingBarChangeListener = ratingBarChangeListener
    }
    private val checkBox = itemView.findViewById<CheckBox>(R.id.foodCheckBox).apply {
        setOnCheckedChangeListener(checkedChangeListener)
    }
    private val textView = itemView.findViewById<TextView>(R.id.labelRatingBar)

    private var cell: RatingBarParamCell? = null

    override fun bind(cell: ParamCell) {
        val ratingBarCell = cell as ParamCell.RatingBar

        ratingBar.apply {
            onRatingBarChangeListener = null
            rating = ratingBarCell.rating
            onRatingBarChangeListener = ratingBarChangeListener
        }

        checkBox.apply {
            if (ratingBarCell.isCheckboxVisible) {
                visibility = View.VISIBLE
                setOnCheckedChangeListener(null)
                isChecked = ratingBarCell.isCheckboxChecked
                setOnCheckedChangeListener(checkedChangeListener)
            } else {
                visibility = View.GONE
            }
        }

        textView.setText(ratingBarCell.labelType.id)
    }
}

class EditTextParamHolder(
    inflater: LayoutInflater,
    parent: ViewGroup,
    listener: ParamAdapter.Listener
) : AbsParamHolder(inflater.inflate(R.layout.rating_bar, parent, false)) {
    // ...
}

class ButtonParamHolder(
    inflater: LayoutInflater,
    parent: ViewGroup,
    listener: ParamAdapter.Listener
) : AbsParamHolder(inflater.inflate(R.layout.rating_bar, parent, false)) {
    // ...
}

class ParamAdapter(
    private var cells: List<ParamCell>,
    private val inflater: LayoutInflater,
    private val listener: Listener
) : RecyclerView.Adapter<AbsParamHolder>() {

    private enum class ViewType {
        RATING_BAR,
        EDIT_TEXT,
        BUTTON
    }

    private val ParamCell.viewType: ViewType
        get() = when (this) {
            is ParamCell.RatingBar -> ViewType.RATING_BAR
            is ParamCell.EditText -> ViewType.EDIT_TEXT
            is ParamCell.Button -> ViewType.BUTTON
        }

    private val viewTypeValues = ViewType.values()

    fun submitList(newCells: List<ParamCell>) {
        cells = newCells
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewTypeOrdinal: Int): AbsParamHolder =
        when (viewTypeValues[viewTypeOrdinal]) {
            ViewType.RATING_BAR -> RatingBarParamHolder(inflater, parent, listener)
            ViewType.EDIT_TEXT -> EditTextParamHolder(inflater, parent, listener)
            ViewType.BUTTON -> ButtonParamHolder(inflater, parent, listener)
        }

    override fun onBindViewHolder(holder: AbsParamHolder, position: Int) =
        holder.bind(cells[position])

    override fun getItemCount(): Int =
        cells.size

    override fun getItemViewType(position: Int): Int =
        cells[position].viewType.ordinal

    interface Listener {
        fun onRatingBarChangeListener(cell: RatingBarParamCell, rating: Float)
        fun onCheckedChangeListener(cell: RatingBarParamCell, isChecked: Boolean)
        // ...
    }
}
```

Note that our cell classes don't know anything about `ViewModel` or `Listener`, `ViewHolder`s simply invoke appropriate `Listener` interface methods, our `ViewModel` can implement this interface, perform some logic and push updated date to the `Fragment` via `LiveData`

So the overall scheme looks like this: 
1. `ViewModel` builds a `List` of cells and puts it into a `LiveData`
1. `Fragment` gets this `List` from the `LiveData` and passes it to an `Adapter`
1. The `Adapter` draws all the cells
1. User clicks something, listener in the `ViewHolder` invokes some method of the `Listener` interface it received in the constructor
1. Our `ViewModel` implements this `Listener` interface. `ViewModel` reacts to the action by creating a copy of our `List`, updating data in it and putting it back to the `LiveData`. And we're back at p.1